pulsar-io/hbase/src/main/java/org/apache/pulsar/io/hbase/sink/HbaseSink.java

```
@@ -0,0 +1,83 @@
+ */
+@Slf4j
+public class HbaseSink extends HbaseAbstractSink<GenericRecord> {
+    @Override
+    public List<Put> bindValue(Record<GenericRecord> message) throws Exception {
+        List<Put> result = new ArrayList<>();
+        GenericRecord record = message.getValue();
+        if (null == record){
+            return null;
+        }
+
+        String rowKeyName = tableDefinition.getRowKeyName();
+        Object rowKeyValue = record.getField(rowKeyName);
+
+        // set familyName value from HbaseSinkConfig
+        String familyName = tableDefinition.getFamilyName();
+        byte[] familyValueBytes = getBytes(familyName);
+
+        List<String> qualifierNames = tableDefinition.getQualifierNames();
+        for (String qualifierName : qualifierNames) {
+            Object qualifierValue = record.getField(qualifierName);
+            if (null != qualifierValue) {
+                Put put = new Put(getBytes(rowKeyValue));
+                put.addColumn(familyValueBytes, getBytes(qualifierName), getBytes(qualifierValue));
+                result.add(put);
+            }
+        }
+
+        return result;
+    }
+
+    private byte[] getBytes(Object value){
+        if (value instanceof Integer) {
+            return Bytes.toBytes((Integer) value);
+        } else if (value instanceof Long) {
+            return Bytes.toBytes((Long) value);
+        } else if (value instanceof Double) {
+            return Bytes.toBytes((Double) value);
+        } else if (value instanceof Float) {
+            return Bytes.toBytes((Float) value);
+        } else if (value instanceof Boolean) {
+            return Bytes.toBytes((Boolean) value);
+        } else if (value instanceof String) {
+            return Bytes.toBytes((String) value);
+        } else if (value instanceof Short) {
+            return Bytes.toBytes((Short) value);
+        } else {
+            //Not support value type, need to add it
+            return null;
```

we should fail fast rather than returning `null`

---

> we should fail fast rather than returning `null`

ok